disnake/state.py

```diff
@@ -1267,25 +1274,32 @@ def parse_guild_member_update(self, data) -> None:
         if member is not None:
             old_member = Member._copy(member)
             member._update(data)
-            user_update = member._update_inner_user(user)
+            user_update = member._update_inner_user(data["user"])
             if user_update:
                 self.dispatch("user_update", user_update[0], user_update[1])
 
             self.dispatch("member_update", old_member, member)
+            user = member
         else:
             if self.member_cache_flags.joined:
                 member = Member(data=data, guild=guild, state=self)
 
                 # Force an update on the inner user if necessary
-                user_update = member._update_inner_user(user)
+                user_update = member._update_inner_user(data["user"])
                 if user_update:
                     self.dispatch("user_update", user_update[0], user_update[1])
 
                 guild._add_member(member)
+                user = member
+            else:
+                user = self.store_user(data["user"])
```

Comment:
Hm, now I'm wondering if this should actually just create a `Member` instead, since we have all the required data anyway.
@onerandomusername thoughts?

Reply:
I think it would be beneficial, imo.